scripts/upgrade

```diff
@@ -113,13 +120,13 @@ then
   else
   ynh_replace_string --match_string="//const WWW_URI = '/asso/';" --replace_string="const WWW_URI = '$path_url/';" --target_file="$final_path/config.local.php"
   fi
-  chown $app:$app $final_path -R
-  chmod 755 $final_path -R
+  chmod 755 $final_path
   # Create the visitors permission if needed
-  if ! ynh_permission_exists --permission "visitors"; then
-    ynh_permission_create --permission "visitors"
+  if [ $is_public -eq 1 ]
+  then
+    ynh_permission_update --permission "main" --add "visitors"
   fi
```

Comment:
is_public is undefined in this context .. what we may want to check is something like : (I'm not sure about the exact helper names / syntax)

```bash
visitors_enabled=$(ynh_permission_has_user "main" "visitors")
if [[ $visitors_enabled -eq 0 ]]
    ynh_permission_update --permission "main" --add "visitors"
fi

# [...] do the curls [...]

if [[ $visitors_enabled -eq 0 ]]
    ynh_permission_update --permission "main" --remove "visitors"
fi
```

Reply:
Ok, also in install script ! A fact is that it should be only a public app, but however it's the right way to use.

Reply:
I can see here from example_ynh it is correct for install script if https://github.com/YunoHost/example_ynh/blob/dd654eb6b82f7928e76324d2e783717681a41f8d/scripts/install#L380

Reply:
Perhaps it is possible just add like the install script in the "Load settings"  `is_public=$YNH_APP_ARG_IS_PUBLIC`

Reply:
Perhaps I am wrong, it may be this ?

```
ynh_permission_has_user --permission=main --user=visitors

if [ $visitors_enabled -eq 0 ]
   ynh_permission_update --permission "main" --remove "visitors"
fi

if [ $visitors_enabled -eq 1 ]
  ynh_permission_update --permission "main" --add "visitors"
fi
```

Reply:
Looking on few apps in the Yunohost-Apps, currently it is use to to do this on upgrade script: 

```
# Cleaning legacy permissions
if ynh_legacy_permissions_exists; then
	ynh_legacy_permissions_delete_all

	ynh_app_setting_delete --app=$app --key=is_public
fi
```
but I don't understand why ?

Reply:
I think it is good now.